packages/jest/src/internals/TestWithPropRunnerBuilder.ts

```diff
@@ -41,6 +57,24 @@ export function buildTestWithPropRunner<Ts extends [any] | any[], TsParameters e
     async () => {
       await fc.assert(propertyInstance, customParams);
     },
-    timeout
+    0x7fffffff // must be 32-bit signed integer
   );
 }
+
+function extractJestGLobalTimeout(): number | undefined {
+  // Initialized via setTimeout, see https://github.com/facebook/jest/blob/fb2de8a10f8e808b080af67aa771f67b5ea537ce/packages/jest-runtime/src/index.ts#L2174
+  const jestTimeout = (globalThis as any)[Symbol.for('TEST_TIMEOUT_SYMBOL')];
+  if (typeof jestTimeout === 'number') {
+    return jestTimeout;
+  }
+  const stateSymbolStringValue = String(Symbol('JEST_STATE_SYMBOL'));
+  for (const key of Object.getOwnPropertySymbols(globalThis)) {
+    if (String(key) === stateSymbolStringValue) {
+      const jestState = (globalThis as any)[key];
+      if (jestState !== null && typeof jestState === 'object' && typeof jestState.testTimeout === 'number') {
+        return jestState.testTimeout;
+      }
+    }
+  }
+  return undefined; // no such case expected
+}
```

Comment:
@SimenB Maybe you'll have a better and longer term suggestion for this hack?!

**What was the problem?** Jest and fast-check had both their timeouts. A common trick I used to give to people was: "use interruptAfterTimeLimit in fast-check with a value slightly inferior to the timeout of Jest". But this trick was damn broken as while fast-check will let's say wait 1000ms and jest 1100ms, there is no certainty that the timeout of fast-check will end before the one of Jest.

**What's the fix about?** In the context of `it.prop` instead of asking jest to handle the timeouts, the handling is fully degated to fast-check, so that there is not risk of overlaps. But to do so I had to find what would be the timeout on jest side... thus this hacky code.

**Is it safe today?** So far, I made it safe by testing it extensively via tests making sure fast-check will always be connected to the expected timeouts. The tests apply various configurations of Jest and timeouts and make sure fast-check extracts the right timeouts.

Reply:
In other words, following this change, when running:

```ts
it.prop([...], async () => {....})
```

The timeout will not be handled by Jest but delegated to fast-check. It will thus be able to interrupt predicate being running and decide whether or not the interrupt was ok or not. But above all it will print the stack, the seed... while today it gets interrupted at the middle and does not output anything.